src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -161,43 +161,40 @@ private static bool UserDefinedConversionIsAllowed(ExpressionSyntax expression,
             //
             // Foo((object)null);
 
-            if (cast.Expression.WalkDownParentheses().IsKind(SyntaxKind.NullLiteralExpression))
```

Comment:
:memo: The diff does a poor job here. The only actual change is the removal of this `if` condition.